### ISSUE TITLE
Secure communication between server, agent and CLI using mTLS

### ISSUE THREAD

**Issue by windsource** [eclipse-ankaios/ankaios#6]
## Description

Currently the communication between server and agent via gRPC is unencrypted. In order to provide authentication and encryption mTLS shall be used.

## Goals

- The mTLS encryption shall be optional (not required during development)
- A script shall be provided to create CA and certificates 
- It shall be documented how to use the generated or existing certificate with Ankaios server and agent
- Server, agent and CLI shall be extended to use mTLS per default (--insecure as extra option)

## Tasks

- [x] Check if existing gRPC crate supports mTLS: yes, an example can be found here: I'll also add the link to the description. Some features are not supported (see [here ](https://github.com/eclipse-ankaios/ankaios/issues/6#issuecomment-2103929156)), but they are irrelevant for us 
- [x] Extend server and agent with mTLS capabilities (also add command line flags) (don't forget to check the permissions on the files and reject starting if the permissions are too open)
- [x] Provide scripts to generate CA and certs
- [x] Enhance documentation
- [x] Enhance SW design and requirement tracing

**Comment by windsource** [comment]
The project https://github.com/eclipse/kuksa.val also uses gRPC between client and databroker via TLS so maybe we can check that before.

**Comment by krucod3** [comment]
As agreed, we first need a concept and we can discuss it together to decide how to proceed.

**Comment by krucod3** [comment]
As proposed by @christoph-hamm, we can use certificates also for scoping (later if needed) the same way Kubernetes does (via the organization and organizational unit fields).

I looked a bit into what tonic supports. There is an example on how to setup mTLS here: https://github.com/hyperium/tonic/tree/master/examples/src/tls_client_auth
and it looks quite straight forward. (I'll also add the link to the description.)

Just a couple of hints:
* From what I see tonic does not support multiple client CAs in parallel (not that we need that).
* Tonic also does not support crl (revocation lists), but we don't need this either.

**Comment by krucod3** [comment]
Also also should check the permissions of the pem and key files read by server and agent and reject starting if the files are readable by group and others.

**Comment by lingnoi** [comment]
The PR with the implementation has been reviewed and merged into main.
